src/scalper.ts

```diff
@@ -574,21 +568,9 @@ Limit: ${hedgePrice} # ${deltaHedgeCount} ID ${deltaOrderId}`);
     jupiter: Jupiter,
   ): Promise<void> {
     // Clean the state by cancelling all existing open orders.
-    const cancelDelta = (
-      await cancelTxOpenBookOrders(this.connection, this.owner, spotMarket, this.symbol));
-    if (cancelDelta !== undefined) {
-      try {
-        await sendAndConfirmTransaction(this.connection, setPriorityFee(cancelDelta), [this.owner]);
-      } catch (err) {
-        console.log(this.symbol, 'Cancel OpenBook Orders', err, err.stack);
-      }
-    }
+    await cancelOpenBookOrders(this.connection, this.owner, spotMarket, this.symbol);
 
-    try {
```

Comment:
These try catch usages in the delta hedge step are okay to me. We'd prefer to have multiple orders in - Haven't tested though. If when this fails it just iterates to the next delta hedge pass thru that's fine. I believe it will just move onto gamma scalping though which would be undesirable

Reply:
Higher risk of needing to delta hedge a big DIP trade and then we fail on a cancel or settle and have to ride out the delta until the next rerun. Better design would be to rerun that specific entire step, delta or gamma when there is any failure. If that isn't bad recursive design, then let's pull the try/catch changes out and aim to do that separately?

I would be less comfortable scaling up next week knowing this is the behavior

Reply:
That risk of a large delta hedge is likely less than having a similar overshoot and being in the same problem or worse if we continue on as we currently do.

We can do a retry with exponential backoff if necessary. But ultimately, we should not proceed if cancel does not work. Settle should be alright to fail open.

There are a few design antipatterns here. By just rerunning the step, either we continue and would have significantly higher likelihood of getting stuck in cascading failure or just get stuck on cancel retries and lose the timing predictability that we rely upon for the scheduler. Once you get into a blocking failure that cannot be cleanly recovered, it is always best to clean up as much as possible then fail hard and fast. Since we have do not have the ability to cancel, it certainly will be worse to fail open and continue.

Planning to get to fixing the design patterns once I get through the more basic fixes in risk-manager.

Reply:
Sounds good trust your judgement